nft/minting_confirmation_task.go

```diff
@@ -0,0 +1,131 @@
+package nft
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/centrifuge/gocelery"
+
+	"github.com/centrifuge/go-centrifuge/centerrors"
+	"github.com/centrifuge/go-centrifuge/queue"
+	"github.com/centrifuge/go-centrifuge/utils"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+)
+
+const (
+	MintingConfirmationTaskName string = "MintingConfirmationTaskName"
+	TokenIDParam                string = "TokenIDParam"
+	BlockHeight                 string = "BlockHeight"
+)
+
+// PaymentObligationMintedFilterer filters the approved NFTs
+type PaymentObligationMintedFilterer interface {
+
+	// FilterPaymentObligationMinted filters PaymentObligationMinted events
+	FilterPaymentObligationMinted(opts *bind.FilterOpts) (*EthereumPaymentObligationContractPaymentObligationMintedIterator, error)
+}
+
+// MintingConfirmationTask confirms the minting of a payment obligation NFT
+type MintingConfirmationTask struct {
+	TokenID                         string
+	BlockHeight                     uint64
+	EthContextInitializer           func() (ctx context.Context, cancelFunc context.CancelFunc)
+	EthContext                      context.Context
+	PaymentObligationMintedFilterer PaymentObligationMintedFilterer
+}
+
+func NewMintingConfirmationTask(
+	nftApprovedFilterer PaymentObligationMintedFilterer,
+	ethContextInitializer func() (ctx context.Context, cancelFunc context.CancelFunc),
+) *MintingConfirmationTask {
```

Comment:
Lets make this CeleryTask instead. We dont need to expose the implementation

Reply:
We must not use the library structs unless its a must, I want to keep the tasks as independent as possible from the queuing library.

Reply:
It is not a struct but rather an Interface MintingConfirmationTask implements

Reply:
thanks, updated my comment.